raiden/src/utils/types.ts

```diff
@@ -143,11 +128,15 @@ export function UInt<S extends number = number>(size?: S) {
   if (siz in sizedCodecCache && name in sizedCodecCache[siz])
     return sizedCodecCache[siz][name] as t.BrandC<typeof BigNumberC, UIntB<S>>;
   if (!(siz in sizedCodecCache)) sizedCodecCache[siz] = {};
-  const max = siz ? bigNumberify(2).pow(siz * 8) : undefined;
+  const max = siz
+    ? bigNumberify(2)
+        .pow(siz * 8)
+        .sub(1)
+    : undefined;
   return (sizedCodecCache[siz][name] = t.brand(
     BigNumberC,
     (n): n is t.Branded<BigNumber, UIntB<S>> =>
-      BigNumberC.is(n) && n.gte(0) && (max === undefined || n.lt(max)),
+      BigNumberC.is(n) && n.gte(0) && (max === undefined || n.lte(max)),
```

Comment:
Why this change? Isn't both the same, as the numbers are always integers anyway? Also, when dealing with ranges due to bits limitation, is more common intervals of type `[a, b[`, therefore `0 <= n < max` is more common.

Reply:
When size is 32 the max will overflow BigNumber and the check will fail.

Reply:
Oh, right, understood your thinking.
Well, it shouldn't happen here, BigNumbers only overflow if you try to construct it with a number which lost precision, otherwise they're kept as strings and BigNumber objects can actually be as large as the memory.
So, these `size` checks are just a compile time validation. At runtime they're both BigNumbers which shouldn't care for size and always test correctly.